### PR TITLE
[main] Argon integration

### PR DESCRIPTION
**Pull request by Tobianas**
JIRA: LIGHTY-108

### CODE REVIEW

File: lighty-examples/lighty-community-restconf-actions-app/src/test/java/io/lighty/examples/controller/actions/RestconfActionsAppTest.java [hunk 1]
```diff
@@ -54,6 +54,15 @@ public static void init() {
     @Test
     public void simpleApplicationTest() throws Exception {
         HttpResponse<String> operations;
+        restClient.POST("restconf/data/network-topology:network-topology/topology=topology-netconf",
+                """
+                        {
```
Comment (ihrasko) [inline lighty-examples/lighty-community-restconf-actions-app/src/test/java/io/lighty/examples/controller/actions/RestconfActionsAppTest.java:59]: Shift this block 4 chars back.

File: lighty-examples/lighty-community-restconf-netconf-app/src/test/java/io/lighty/examples/controllers/restconfapp/tests/RestconfAppTest.java [hunk 2]
```diff
@@ -46,6 +46,16 @@ public static void init() {
     @Test
     public void simpleApplicationTest() throws IOException, InterruptedException {
         HttpResponse<String> operations;
+        restClient.POST("restconf/data/network-topology:network-topology/topology=topology-netconf",
+                """
+                        {
```
Comment (ihrasko) [inline lighty-examples/lighty-community-restconf-netconf-app/src/test/java/io/lighty/examples/controllers/restconfapp/tests/RestconfAppTest.java:51]: Shift this block 4 chars back.

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfUtils.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ private NetconfUtils() {
     public static InstanceIdentifier<Node> createNetConfNodeMountPointII(final NodeId nodeId) {
         KeyedInstanceIdentifier<Topology, TopologyKey> instanceIdentifier =
                 InstanceIdentifier.create(NetworkTopology.class).child(Topology.class,
-                        new TopologyKey(new TopologyId(TopologyNetconf.QNAME.getLocalName())));
+                        new TopologyKey(new TopologyId(NetconfNodeUtils.DEFAULT_TOPOLOGY_NAME)));
```
Comment (ihrasko) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfUtils.java:76]: Revert this back, `NetconfNodeUtils.DEFAULT_TOPOLOGY_NAME` is marked deprecated and will be removed.

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 4]
```diff
@@ -84,12 +86,11 @@ protected boolean initProcedure() {
...
         this.schemaCtxHandler
                 = new SchemaContextHandler(this.domDataBroker, this.domSchemaService);
-        this.schemaCtxHandler.init();
         final Configuration streamsConfiguration = RestConfConfigUtils.getStreamsConfiguration();
 
         LOG.info("Starting RestconfApplication with configuration {}", streamsConfiguration);
-
-        final RestconfApplication restconfApplication = new RestconfApplication(schemaCtxHandler,
+        final DatabindProvider provider = new DOMDatabindProvider(domSchemaService);
+        final RestconfApplication restconfApplication = new RestconfApplication(provider,
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:93]: ignore

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 5]
```diff
@@ -84,12 +86,11 @@ protected boolean initProcedure() {
...
                 this.domMountPointService, this.domDataBroker, this.domRpcService, this.domActionService,
                 this.domNotificationService, this.domSchemaService, streamsConfiguration);
         final ServletContainer servletContainer8040 = new ServletContainer(ResourceConfig
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:96]: ignore

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 6]
```diff
@@ -84,12 +86,11 @@ protected boolean initProcedure() {
 
         this.schemaCtxHandler
                 = new SchemaContextHandler(this.domDataBroker, this.domSchemaService);
-        this.schemaCtxHandler.init();
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:87]: We can remove `schemaContextHandler` completely from this class.

File: lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml [hunk 7]
```diff
@@ -72,6 +72,10 @@
             <artifactId>mockito-inline</artifactId>
             <scope>test</scope>
         </dependency>
+      <dependency>
```
Comment (ihrasko) [inline lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml:75]: What's the purpose of this change? I need to recheck our dependencies...

File: lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java [hunk 8]
```diff
@@ -45,7 +45,7 @@ public void loadTopLevelClusterModelsFromJsonConfig() throws Exception {
                 .getEffectiveModelContext().getModules().size();
         assertTrue(lightyController.shutdown(TIME_OUT, TimeUnit.SECONDS));
 
-        assertEquals(loadedModulesSize, 17);
+        assertEquals(loadedModulesSize, 15);
```
Comment (ihrasko) [inline lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java:48]: No, you have to adapt revisions in: `sampleConfigCluster.json`.

File: lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java [hunk 9]
```diff
@@ -33,7 +33,7 @@ public void loadTopLevelSingleNodeModelsFromJsonConfig() throws Exception {
                 .getEffectiveModelContext().getModules().size();
         assertTrue(lightyController.shutdown(TIME_OUT, TimeUnit.SECONDS));
 
-        assertEquals(loadedModulesSize, 16);
```
Comment (ihrasko) [inline lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java:36]: No, you have to adapt revisions in: `sampleConfigSingleNode.json`.

File: lighty-core/dependency-versions/pom.xml [hunk 10]
```diff
@@ -129,7 +129,7 @@
             <dependency>
                 <groupId>org.junit</groupId>
                 <artifactId>junit-bom</artifactId>
```
Comment (ihrasko) [inline lighty-core/dependency-versions/pom.xml:131]: Let's remove this commit for now. It will be part of another PR.

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-core/lighty-binding-parent/pom.xml [hunk 11]
```diff
@@ -49,12 +49,12 @@
             <plugin>
                 <groupId>org.opendaylight.yangtools</groupId>
                 <artifactId>yang-maven-plugin</artifactId>
-                <version>9.0.1</version>
+                <version>10.0.2</version>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-core/lighty-binding-parent/pom.xml]: fixed

File: lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/dto/NetconfDeviceResponse.java [hunk 12]
```diff
@@ -10,19 +10,19 @@
...
-import org.opendaylight.yang.gen.v1.urn.opendaylight.netconf.node.topology.rev150114.NetconfNodeConnectionStatus.ConnectionStatus;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.netconf.device.rev221225.ConnectionOper;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.netconf.node.topology.rev221225.NetconfNode;
 import org.opendaylight.yang.gen.v1.urn.tbd.params.xml.ns.yang.network.topology.rev131021.network.topology.topology.Node;
 
 public class NetconfDeviceResponse {
 
     private final String nodeId;
-    private final ConnectionStatus connectionStatus;
+    private final ConnectionOper.ConnectionStatus connectionStatus;
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-examples/lighty-controller-springboot-netconf/src/main/java/io/lighty/core/controller/springboot/rest/dto/NetconfDeviceResponse.java]: changed

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java (hunk 6 above)
Comment (Tobianas) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:87]: removed in 7e95206af83546f98cd62b0d592c8b4705566654

File: lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml (hunk 7 above)
Comment (Tobianas) [reply lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml:75]: as stated in c5936074d63b807ccd8852975ed7b4ba1e85476c This import is required for AuthenticationManager.java, HttpBasicAuth.java and TokenAuthenticators.java used in AAALightyShiroProvider

File: lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java (hunk 8 above)
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java]: removed this commit and changed back to original size.

File: lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java (hunk 9 above)
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/SampleConfigTest.java]: removed this commit and changed back to original size.

File: lighty-core/dependency-versions/pom.xml (hunk 10 above)
Comment (Tobianas) [reply lighty-core/dependency-versions/pom.xml]: commit droped

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/LightyDOMActionService.java [hunk 13]
```diff
@@ -32,7 +32,7 @@
 import org.opendaylight.yangtools.yang.data.api.schema.ContainerNode;
 import org.opendaylight.yangtools.yang.model.api.stmt.SchemaNodeIdentifier.Absolute;
 
-public final class LightyDOMActionService implements DOMActionService {
+public final class LightyDOMActionService implements RemoteDeviceServices.Actions.Normalized {
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/LightyDOMActionService.java]: Fixed

File: lighty-examples/lighty-community-restconf-actions-app/src/test/java/io/lighty/examples/controller/actions/RestconfActionsAppTest.java (hunk 1 above)
Comment (Tobianas) [reply lighty-examples/lighty-community-restconf-actions-app/src/test/java/io/lighty/examples/controller/actions/RestconfActionsAppTest.java:59]: Shifted 4chars back

File: lighty-examples/lighty-community-restconf-netconf-app/src/test/java/io/lighty/examples/controllers/restconfapp/tests/RestconfAppTest.java (hunk 2 above)
Comment (Tobianas) [reply lighty-examples/lighty-community-restconf-netconf-app/src/test/java/io/lighty/examples/controllers/restconfapp/tests/RestconfAppTest.java:51]: Shifted 4chars back

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java (hunk 6 above)
Comment (Tobianas) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:87]: basically removing this code causes restconf to break: 
`new SchemaContextHandler(this.domDataBroker, this.domSchemaService);`

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 14]
```diff
@@ -81,15 +82,12 @@ public CommunityRestConf(final DOMDataBroker domDataBroker,
     @Override
     protected boolean initProcedure() {
         final Stopwatch stopwatch = Stopwatch.createStarted();
-
-        this.schemaCtxHandler
-                = new SchemaContextHandler(this.domDataBroker, this.domSchemaService);
-        this.schemaCtxHandler.init();
         final Configuration streamsConfiguration = RestConfConfigUtils.getStreamsConfiguration();
+        new SchemaContextHandler(this.domDataBroker, this.domSchemaService);
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:86]: ignore

**Review by ihrasko — APPROVED** [review]
